src/gui/sdlmain.cpp

```diff
@@ -2923,28 +2923,48 @@ static void show_warning(char const * const message) {
 #endif // WIN32
 }
 
-static void launcheditor() {
-	std::string path,file;
+static int LaunchEditor()
+{
+	std::string path, file;
 	Cross::CreatePlatformConfigDir(path);
 	Cross::GetPlatformConfigName(file);
 	path += file;
-	FILE* f = fopen(path.c_str(),"r");
+
+	FILE *f = fopen(path.c_str(), "r");
 	if (!f && !control->PrintConfig(path)) {
-		printf("tried creating %s. but failed.\n", path.c_str());
-		exit(1);
+		fprintf(stderr, "Tried creating '%s', but failed.\n", path.c_str());
+		return 1;
 	}
-	if(f) fclose(f);
-/*	if(edit.empty()) {
-		printf("no editor specified.\n");
-		exit(1);
-	}*/
-	std::string edit;
-	while(control->cmdline->FindString("-editconf",edit,true)) //Loop until one succeeds
-		execlp(edit.c_str(),edit.c_str(),path.c_str(),(char*) 0);
-	//if you get here the launching failed!
-	printf("can't find editor(s) specified at the command line.\n");
-	exit(1);
+	if (f)
+		fclose(f);
+
+	auto replace_with_process = [path](const std::string &prog) {
+		execlp(prog.c_str(), prog.c_str(), path.c_str(), (char *)nullptr);
+	};
+
+	std::string editor;
+	constexpr bool remove_arg = true;
+	// Loop until one succeeds
+	while (control->cmdline->FindString("--editconf", editor, remove_arg))
+		replace_with_process(editor);
+
+	while (control->cmdline->FindString("-editconf", editor, remove_arg))
+		replace_with_process(editor);
+
+	const char *env_editor = getenv("EDITOR");
+	if (env_editor)
+		replace_with_process(env_editor);
+
+	replace_with_process("nano");
+	replace_with_process("vim");
+	replace_with_process("vi");
+	replace_with_process("notepad.exe");
+
+	fprintf(stderr, "Can't find any text editors.\n");
+	fprintf(stderr, "You can override EDITOR env variable to use your preffered one.\n");
```

Comment:
You can set the EDITOR env variable to your preferred text editor.

Reply:
Changed :)